misc/python/materialize/parallel_workload/executor.py

```diff
@@ -151,7 +142,11 @@ def execute(
                         raise QueryError(str(e), query)
                 else:
                     try:
-                        self.cur.execute(query.encode())
+                        if query == "commit":
```

Comment:
This looks a bit brittle. What if it is "commit;"? Or is it only invoked from the function changed above?

Reply:
Only invoked from the function changed above. The problem is that sometimes we'd still run a `self.cur.execute("commit")`, which is not supported in psycopg.